operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ResourceConfiguration.java

```diff
@@ -111,11 +111,12 @@ static Set<String> ensureValidNamespaces(Collection<String> namespaces) {
    *
    * @return a Set of namespace names the associated controller will watch
    */
-  default Set<String> getEffectiveNamespaces(ConfigurationService configurationService) {
+  default Set<String> getEffectiveNamespaces(ControllerConfiguration<?> controllerConfiguration) {
```

Comment:
Yes, but  this is just an utility method effectively. I think the problem is more about that `ControllerConfiguraiton` extends `ResourceConfiguration`, since Controller is not really a resource resource. It should rather reference a resource configuration (the primary resource), but not sure if we want to change that at this point.

Maybe in a different PR we can take a look on this.